ui/app/AppLayouts/Profile/stores/ContactsStore.qml

```diff
@@ -121,6 +138,7 @@ QtObject {
     }
 
     function requestContactInfo(publicKey) {
```

Comment:
What If we call this for two contacts? We're probably still have an issue with one of them.

Reply:
my suggestion would be to move this `d.requestedContactPubKey` to the `SendContactRequestModal` instead and then the `store` just to forward the nim signal.

Reply:
> What If we call this for two contacts? We're probably still have an issue with one of them.

I guess the only way that could happen is if we close the current CR popup and open a new one with someone else, in which case the previous one will never send a signal, but that's fine because the previous modal is closed anyway.

I guess it would make the store code cleaner. Does it matter that the Connections stays active at all times though?

Reply:
I agree with @alexjba. That's the solution inline with the guide. Stores should just forward what's exposed via context properties, no logic here as it's kind of "dead zone" for ui side unit tests.

So contacts store should expose `signal contactInfoRequestFinished(...)`. Forwarding signal can be done e.g. in `Component.onCompleted` as `d.contactsModuleInst.contactInfoRequestFinished.connect(root.contactInfoRequestFinished)`

And the rest of logic goes to the modal.

Reply:
Thanks for the review @alexjba @micieslak 

I updated according to your suggestions. I also removed the use of `getContactAsJSON` and used `SQUtils.ModelUtils.getByKey` instead. It all works as expected!